extension/src/background/websocket-client.ts

```diff
@@ -0,0 +1,32 @@
+export default class WebSocketClient {
+    private host: string;
+    private webSocket!: WebSocket;
+
+    public constructor(host: string) {
+        this.host = host;
+    }
+
+    public getWebSocket(): WebSocket {
+        return this.webSocket;
+    }
+
+    public connect(): Promise<void> {
+        return new Promise((resolve) => {
+            this.webSocket = new WebSocket(this.host);
+
+            this.webSocket.onopen = () => {
+                this.webSocket.send(
+                    JSON.stringify({ message: "[WS] Connected to server" })
```

Comment:
```suggestion
                    JSON.stringify({ message: "[EWS] Connected to server" })
```
What about using [EWS] for Extension WebSocket andd [SWS] for ServerWebSocket (or maybe [WS-E] and [WS-S]